airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py

```diff
@@ -137,10 +142,21 @@ def infer_schema_process(
         # and so we can't multiprocess with the actual fileobject on Windows systems
         # we're reading block_size*2 bytes here, which we can then pass in and infer schema from block_size bytes
         # the *2 is to give us a buffer as pyarrow figures out where lines actually end so it gets schema correct
-        file_sample = file.read(self._read_options()["block_size"] * 2)
-        schema_dict = self._run_in_external_process(
-            infer_schema_process, 4, 60, file_sample, self._read_options(), self._parse_options(), self._convert_options()
-        )
+        if self._format.get("infer_datatypes", True):
+            self.logger.debug("inferring schema")
+            file_sample = file.read(self._read_options()["block_size"] * 2)
+            schema_dict = self._run_in_external_process(
+                infer_schema_process, 4, 60, file_sample, self._read_options(), self._parse_options(), self._convert_options()
+            )
```

Comment:
please call functions like this with kwargs arguments, it is very difficult to understand what meaning of each parameter.
Also I don't think neither `infer_schema_process` nor `infer_schema_process` belongs to this class, if you don't mind please move it outside, next to `multiprocess_queuer`. In such way we can easily test it.